lib/diagnostic.js

```diff
@@ -14,12 +14,30 @@
 'use strict';
 
 const fs = require('fs'); // jshint ignore: line
+const stdin = '/dev/stdin';
 
-const sumLines = (filename, callback) => {
-  /* your response here */
-  callback();
+let inFile = process.argv[2] === '-' ? stdin : process.argv[2];
+
+const sumLines = (array /*filename, callback*/) => {
+
+  let result;
+  for (let i = 0; i < array; i++) {
```

Comment:
You have a for-loop in conjunction with recursion. That is never the right thing to do. The whole point of a loop is to do something repeatedly. And that is also the strategy behind recursion.